### PR TITLE
SPIN-3116 custom random password setter

### PR DESCRIPTION
**Pull request by samongyr-sq**
Allow the charset and password length for the random password setter to be configurable

### CODE REVIEW

File: secret.go [hunk 1]
```diff
@@ -57,8 +57,26 @@ const (
...
-type RandomPassword struct{}
+// it only supports rotation by Secrets Manager. The password generated by
+// RandomPassword may be configured by providing RandomPasswordOptions on
+// initialization.
+type RandomPassword struct {
+	Opts *RandomPasswordOptions
+}
+
+// RandomPasswordOptions is used to configure a RandomPassword
+type RandomPasswordOptions struct {
```
Comment (daniel-nichter) [inline secret.go]: I think these options can/should be fields in `RandomPassword`. Option structs are usually used as params to func() calls, or parts of struct/objs that have a lot of fields and opts are just one small part of the exported fields. 

Example: https://golang.org/pkg/net/http/#Server some of the fields in that are the options; they configure the object. Similarly, these fields configure the `RandomPassword`.

File: secret.go [hunk 2]
```diff
@@ -57,8 +57,26 @@ const (
...
+	// PasswordLength defines the length of the password generated by RandomPassword.
+	// If not provided, DEFAULT_PASSWORD_LENGTH will be used
```
Comment (daniel-nichter) [inline secret.go:67]: I don't see `DEFAULT_PASSWORD_LENGTH` defined/declared in this PR? Missing a file?

Comment (daniel-nichter) [reply secret.go:67]: Or did it already exist? (I don't recall it.)

Comment (samongyr-sq) [reply secret.go:67]: already existed but unused heh https://github.com/square/password-rotation-lambda/blob/master/secret.go#L49

File: secret.go (hunk 1 above)
Comment (samongyr-sq) [reply secret.go]: I think separating configurable options from the other components in a separate defined struct is cleaner, especially for larger more complex structs, but I don't feel super strongly about it/don't have a good enough reason so I made the changes.

Comment (samongyr-sq) [reply secret.go]: oh, I remember why I like it --- makes it super easy to pass in YAML configs

File: secret.go [hunk 3]
```diff
@@ -57,8 +57,23 @@ const (
...
-type RandomPassword struct{}
+// it only supports rotation by Secrets Manager. The password generated by
+// RandomPassword may be configured by setting either `PasswordLength` or
+// `ValidCharset` on initialization
+type RandomPassword struct {
+	// Options to configure the random password generated.
+
+	// PasswordLength defines the length of the password generated by RandomPassword.
+	// If not provided, DEFAULT_PASSWORD_LENGTH will be used
+	PasswordLength *int
```
Comment (daniel-nichter) [inline secret.go]: Avoid pointers to primitive in Go. Idiomatic Go just uses zero values. In this case, it's fair to say that a password len cannot be zero chars long, so if not specified (zero value) then the default is used.

**Review by daniel-nichter — APPROVED** [review]

**Pull request merged**